app/controllers/fhir/r4/api_controller.rb

```diff
@@ -298,14 +298,16 @@ def create
           status_unprocessable_entity(resource, fhir_map, req_json) && return
         end
 
-        Rails.logger.info "Created Patient with ID: #{resource.id}"
-
-        # Send enrollment notification only to responders
-        resource.send_enrollment_notification if resource.self_reporter_or_proxy?
-
         # Create a history for the enrollment
         History.enrollment(patient: resource, created_by: @current_actor_label, comment: 'Monitoree enrolled via API.')
       end
+
+      # This is necessary since the transaction will swallow an ActiveRecord::Rollback error
+      raise(StandardError, 'Error when saving Patient to the database') if resource.id.nil?
```

Comment:
So is this line necessary in case we do add explicit rolling back, with `raise ActiveRecord::Rollback`. I can only get to this line of code if I add a line like that within the `.transaction` block. That makes sense if it's the case, but I want to make sure I'm not misunderstanding or missing something.

Reply:
Not really for in case _we_ add an explicit rollback, but rather in case ActiveRecord raises that error for whatever reason when saving to the database. I am not aware of any specific way to trigger that error, and so it's just there "just in case" and not an expected case.

Reply:
Sounds good. My reading of this doc [here](https://api.rubyonrails.org/v6.1.0/classes/ActiveRecord/Rollback.html) makes me think that generally that error wouldn't be raised unless we do it ourselves. But it's not 100% clear to me, and it's possible in the future we would add raising of that error and not realize it.